backend/main.py

```
@@ -38,30 +74,183 @@
+    """
+    # Validate uploaded file
+    await validate_uploaded_file(image)
 
-        # 3. Start Telegram Bot in separate thread
-        await run_in_threadpool(start_bot_thread)
-        logger.info("Telegram bot started in separate thread.")
+    # Convert to PIL Image directly from file object to save memory
+    try:
+        pil_image = await run_in_threadpool(Image.open, image.file)
+        # Validate image for processing
+        await run_in_threadpool(validate_image_for_processing, pil_image)
```

Ruff (0.14.14)

[error] Undefined name `validate_image_for_processing`

(F821)